modules/metal-connection/main.tf

```diff
@@ -0,0 +1,53 @@
+data "equinix_fabric_service_profiles" "zside" {
+  count = var.zside_ap_type == "SP" ? 1 : 0
+  filter {
+    property = "/name"
+    operator = "="
+    values   = [var.zside_fabric_sp_name]
+  }
+}
+
+resource "equinix_fabric_connection" "primary_metal_connection" {
+  name = var.connection_name
+  type = var.connection_type
+  notifications {
+    type   = var.notifications_type
+    emails = var.notifications_emails
+  }
+  dynamic "project" {
+    for_each = var.project_id != "" ? [1] : []
+    content {
+      project_id = var.project_id
+    }
+  }
+  additional_info = var.additional_info != [] ? var.additional_info : null
+  bandwidth       = var.bandwidth
+  order {
+    purchase_order_number = var.purchase_order_number != "" ? var.purchase_order_number : null
+  }
+  a_side {
+    access_point {
+      type               = var.aside_ap_type
```

Comment:
If this is truly a module for Metal then the Access Point will always be the same, right?